dev/archery/tests/generate_files_for_endian_test.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+: ${ARROW_DIR:=/arrow}
+
+json_file=`archery integration --with-cpp=1 | grep "Testing file" | tail -1 | awk '{ print $3 }'`
```

Comment:
It's not obvious to me what this script is doing precisely. In any case, I don't think we should rely on precise human-readable output of `archery integration`.

Reply:
Lines 21 and 22 try to get the directory where json files are generated. As far as I know, there is no way to get the directory name except stdout of `archery integration`.

Reply:
`archery integration --tempdir` should allow you to choose the directory.

Reply:
Thank you for your good suggestion. I will try it.